gtsfm/averaging/rotation/shonan.py

```diff
@@ -100,6 +104,14 @@ def run(self, num_images: int, i2Ri1_dict: Dict[Tuple[int, int], Optional[Rot3]]
                 underconstrained system or ill-constrained system), or where the camera pose had no valid observation
                 in the input to run().
         """
+        if len(i2Ri1_dict) == 0:
+            logger.warning(
```

Comment:
warning or error?

Reply:
Thanks Ayush. Either would work, but I prefer warning since it may be expected behavior for an impossibly challenging dataset.